node/src/store.rs

```diff
@@ -135,4 +163,68 @@ pub enum StoreError {
 
     #[error("Error opening store: {0}")]
     OpenFailed(String),
+
+    #[error("Invalid headers range")]
+    InvalidHeadersRange,
+}
+
+/// a helper function to convert any kind of range to the inclusive range of header heights.
+fn to_headers_range(bounds: impl RangeBounds<u64>, last_index: u64) -> Result<RangeInclusive<u64>> {
+    let start = match bounds.start_bound() {
+        Bound::Unbounded => 1,
+        Bound::Included(&x) => x,
+        Bound::Excluded(&u64::MAX) => return Err(StoreError::InvalidHeadersRange),
+        Bound::Excluded(&x) => x + 1,
+    };
+    let end = match bounds.end_bound() {
+        Bound::Unbounded => last_index,
+        Bound::Included(&x) => x,
+        Bound::Excluded(&0) => return Err(StoreError::InvalidHeadersRange),
```

Comment:
I think instead of this (and the one in `start_bound` match). We should do `if start > end`

Reply:
but then we will already hit overflow

Reply:
Ok true, didn't thought it before.

Can you add the `if start > end` after them?

Reply:
refactored this part. I haven't included this condition as per offline discussion